### ISSUE TITLE
[WP 5.3] Declaration should be compatible with `Walker::walk`

### ISSUE THREAD

**Issue by Chouby** [Automattic/VIP-Coding-Standards#448]
## Bug Description

PHP 5.6 or above
WP 5.3-beta3
VIPCS 2.0

WP 5.3 modified the signature of `Walker::walk()`. If we adapt our code to this new signature, we get an error:
```
Declaration of `PLL_Walker_List::walk($elements, $max_depth, $args)` should be compatible with `Walker::walk($elements, $max_depth)`
(WordPressVIPMinimum.Classes.DeclarationCompatibility.DeclarationCompatibility)
```
The sniff should be adpated to the new method signature

## Minimal Code Snippet

```php
class PLL_Walker_List extends Walker {
	public function walk( $elements, $max_depth, ...$args ) {
		return parent::walk( $elements, $max_depth, $args );
	}
}
```

## Tested Against `master` branch?

No but the code hasn't been modified for 5 months so cannot take this WP modification into account.

**Comment by GaryJones** [comment]
Thanks for reporting @Chouby!

Relevant lines: https://github.com/Automattic/VIP-Coding-Standards/blob/develop/WordPressVIPMinimum/Sniffs/Classes/DeclarationCompatibilitySniff.php#L148-L151

Changed signature in Walker:walk(): https://github.com/WordPress/WordPress/commit/38a8513cf2a58c9bbc0a48aff5868c919fde6ef7

**Comment by jrfnl** [comment]
@Chouby Just a little heads-up, your code sample is bug-prone.

The correct way to do it would be:
```php
class PLL_Walker_List extends Walker {
	public function walk( $elements, $max_depth, ...$args ) {
		return parent::walk( $elements, $max_depth, ...$args );
	}
}
```

Note the argument unpacking in the return statement. If you pass `$args` to the parent without that, you are passing an array, not the individual arguments received.

**Comment by jrfnl** [comment]
Related: https://make.wordpress.org/core/2019/10/09/wp-5-3-introducing-the-spread-operator/

**Comment by Chouby** [comment]
@jrfnl Thank you! This indeed could have been a bug. I removed most of the original code where at some point `$args =  $args[0]` to construct the minimal code snippet, when I should have built a new one to avoid making this mistake ;-)

**Comment by jrfnl** [comment]
@Chouby No worries, I just pointed it out in hopes of preventing the issue getting into a real codebase (or others finding this snippet and using it blindly).

**Comment by jrfnl** [comment]
@Chouby PR #505 should fix this. Testing appreciated.
